### PR TITLE
Migrate login command to new client

### PR DESCRIPTION
**Pull request by marians**
This continues the client migration and replaces the legacy Login function.

Replaces #253

### CODE REVIEW

**Review by oponder — APPROVED** [review]
LGTM, but have some questions

File: client/clienterror/clienterror.go [hunk 1]
```diff
@@ -59,16 +59,33 @@ func New(err error) *APIError {
 	// We first handle the most specific cases, which differ between operations.
 	// When adding support for more API operations to the client, add handling
 	// of any new specific error types here.
-	convertedError, ok := err.(*auth_tokens.DeleteAuthTokenUnauthorized)
+	deleteAuthoTokenUnauthorizedError, ok := err.(*auth_tokens.DeleteAuthTokenUnauthorized)
```
[the remark this answers is not in the document]
Comment (marians) [reply client/clienterror/clienterror.go]: No. Would you prefer `Auto`? ;-) will fix

[the remark this answers is not in the document]
Comment (marians) [reply client/clienterror/clienterror.go]: done

File: client/clienterror/clienterror.go [hunk 2]
```diff
@@ -59,16 +59,33 @@ func New(err error) *APIError {
...
 		return &APIError{
 			HTTPStatusCode: http.StatusUnauthorized,
-			OriginalError:  convertedError,
+			OriginalError:  deleteAuthoTokenUnauthorizedError,
 			ErrorMessage:   "Unauthorized",
 			ErrorDetails:   "The token presented was likely no longer valid. No further action is required.",
 		}
 	}
 
+	createAuthoTokenUnauthorizedError, ok := err.(*auth_tokens.CreateAuthTokenUnauthorized)
```
[the remark this answers is not in the document]
Comment (marians) [reply client/clienterror/clienterror.go]: done

File: commands/login_giantswarm.go [hunk 3]
```diff
@@ -28,10 +24,13 @@ func loginGiantSwarm(args loginArguments) (loginResult, error) {
 		result.endpointSwitched = true
 	}
 
-	encodedPassword := base64.StdEncoding.EncodeToString([]byte(args.password))
```
[the remark this answers is not in the document]
Comment (marians) [reply commands/login_giantswarm.go]: The wrapper function does it, see https://github.com/giantswarm/gsctl/blob/migrate-login-2/client/client.go#L187

**Comment by marians** [comment]
thx!

**Pull request merged**